### PR TITLE
fix getnericPLEG still return an empty podIP when get podIP from older cached status

### LINKED ISSUE #82276: kubelet: getnericPLEG still return an empty podIP when get podIP from older cached status
**What happened**:
Nowadays, when pod dead and its network tear down, PLEG will get pod status from underlying runtime(e.g. docker) with an empty pod IP address. then kubelet tries to get podIP from *PLEG's runtime states cache* and use this pod's old status' IP address, so that the pod IP address will be consistent at its whole lifecycle.

https://github.com/kubernetes/kubernetes/blob/c7c89f8c6140e0639dc8ba2d8dcdd6fdbd5409ae/pkg/kubelet/pleg/generic.go#L396-L406

However in `GenericPLEG.getPodIPs` method, we will iterate all sandbox containers, If one of those sandboxs' state is READY, we will return `status.IP` directly even though we do not touch this `status.IP` at all and it still is empty.
 
https://github.com/kubernetes/kubernetes/blob/c7c89f8c6140e0639dc8ba2d8dcdd6fdbd5409ae/pkg/kubelet/pleg/generic.go#L349-L382

I don't know why we return an empty podIP directly and which seems opposed to the original intention of this function to prevent an empty podIP.

On the other hand, even though the sandbox status is READY, the sandbox's IP may still is empty, as when we stop a pod,1) we first tear down network and mark the network as not ready, 2) then we stop sandbox container. Between those two steps, we do not have any lock, there are some chances that other goroutines may read container status and get an empty IP from a READY sandbox. so the sandbox's status seems to be irrelevant to podIP.

https://github.com/kubernetes/kubernetes/blob/c7c89f8c6140e0639dc8ba2d8dcdd6fdbd5409ae/pkg/kubelet/dockershim/docker_sandbox.go#L253-L264

there are some workload controllers (e.g. endpoints controller, custom controller) rely heavily on podIP, we should fix this.

**What you expected to happen**:
We shouldn't have the logic of iterating and checking pod sandbox status. when we get an empty podIP from underlying runtime, we should return the cached older status' podIP.

**How to reproduce it (as minimally and precisely as possible)**:
run [this snippet](https://gist.github.com/gaorong/60e84a7534546f56431b9edc46727ffb) and wait a few moments.

**Anything else we need to know?**:
none
**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.0-beta.1", GitCommit:"92639f719ee353219fa2cd19128a93409bb3a1e3", GitTreeState:"clean", BuildDate:"2019-08-20T13:32:12Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.0-beta.1", GitCommit:"92639f719ee353219fa2cd19128a93409bb3a1e3", GitTreeState:"clean", BuildDate:"2019-08-20T13:22:52Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

/assign
/sig node
/area kubelet

### PR DESCRIPTION
**Pull request by gaorong**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #82276

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/kubelet/pleg/generic.go [hunk 1]
```diff
@@ -358,24 +357,6 @@ func (g *GenericPLEG) getPodIPs(pid types.UID, status *kubecontainer.PodStatus)
 		return nil
 	}
```
Comment (xiaoanyunfei) [inline pkg/kubelet/pleg/generic.go]: `|| len(oldStatus.IPs) == 0 ` is unnecessary here

Comment (gaorong) [reply pkg/kubelet/pleg/generic.go]: done, thanks

**Comment by matthyx** [comment]
@gaorong can you add a test that covers this particular case? It should fail without your fix to avoid regression in the future. Thanks!

**Comment by gaorong** [comment]
> can you add a test that covers this particular case? It should fail without your fix to avoid regression in the future. Thanks!

Done, have added test case.

**Pull request closed without merging**